### PR TITLE
[Ldap] Fixed issue with legacy client initialisation

### PR DESCRIPTION
**Pull request by csarrazi**
| Q | A |
| --- | --- |
| Branch? | 3.1 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | https://github.com/symfony/symfony/pull/18725#issuecomment-223076867 |
| License | MIT |
| Doc PR |  |

Thanks @uvups for noticing this.

### CODE REVIEW

**Comment by fabpot** [comment]
Any way to add some tests?

**Comment by csarrazi** [comment]
Sure! I'll try to find some time for this tomorrow

**Comment by uvups** [comment]
@csarrazi  This fix worked for me. Thanks

File: src/Symfony/Component/Ldap/LdapClient.php [hunk 1]
```diff
@@ -24,13 +24,22 @@
 
     public function __construct($host = null, $port = 389, $version = 3, $useSsl = false, $useStartTls = false, $optReferrals = false, LdapInterface $ldap = null)
     {
+        if ((bool) $useSsl) {
```
Comment (stof) [inline src/Symfony/Component/Ldap/LdapClient.php]: no need to cast explicitly here

**Comment by stof** [comment]
btw, this class is missing its deprecation warning

File: src/Symfony/Component/Ldap/LdapClient.php (hunk 1 above)
Comment (csarrazi) [inline src/Symfony/Component/Ldap/LdapClient.php]: True. However this was kept to ensure the previous behaviour.

I can remove the cast, if needed.

**Comment by csarrazi** [comment]
You mean in the constructor, right?

**Comment by nicolas-grekas** [comment]
@csarrazi stof means after the namespace declaration. See `grep '^@trigger' src/ -r`

**Comment by csarrazi** [comment]
Ok. I'll check this tonight! Thanks for the info @nicolas-grekas!

**Comment by csarrazi** [comment]
@nicolas-grekas @stof Done.

File: src/Symfony/Component/Ldap/LdapClient.php [hunk 2]
```diff
@@ -11,6 +11,8 @@
 
 namespace Symfony\Component\Ldap;
 
+@trigger_error('The '.__NAMESPACE__.'\LdapClient class is deprecated since version 3.1 and will be removed in 4.0. Use directly the Ldap class instead.', E_USER_DEPRECATED);
```
Comment (fabpot) [inline src/Symfony/Component/Ldap/LdapClient.php]: `Use the Ldap class directly instead.`

**Comment by nicolas-grekas** [comment]
:+1: 
Just wondering: do the added tests make sense on the Ldap class also? Or will it be OK to just remove them when doing 4.0?

File: src/Symfony/Component/Ldap/LdapClient.php (hunk 2 above)
Comment (csarrazi) [inline src/Symfony/Component/Ldap/LdapClient.php]: Done

**Comment by csarrazi** [comment]
It will be okay to remove them for 4.0. The LdapClientTest tests only happen to test the BC layer:

* whether the Ldap class is properly encapsulated or not in the LdapClient class;
* the BC-compatible `find()` method;
* the config change between the two classes.

**Comment by fabpot** [comment]
Thank you @csarrazi.

**Pull request merged**